### PR TITLE
BEOBLE-1476-Fix-Vercel-staging-env-based-plugin-key-mixup

### PR DESCRIPTION
**Pull request by vanilla-james**

### CODE REVIEW

**Comment by vanilla-james** [comment]
@bamnenim-kryptonium 
I tested the deployment as much as I could without actually deploying to dev (as deploying to dev is blocked)

### How issue was resolved
---
* swtiched `process.env.NEXT_PUBLIC_VERCEL_ENV` to `process.env.VERCEL_ENV` as _document.tsx apparently is run in the server and `process.env.NEXT_PUBLIC_VERCEL_ENV` might not behave as we expect / intended
* `process.env.VERCEL_ENV` should be checked against `production`, `preview`, or `development` - fixed

### Test deployments
---
deployed so that `prod` is shown:
https://beoble-toolkit-9l29p020y-kryptonium.vercel.app/
https://vercel.com/kryptonium/beoble-toolkit/Jq9ydAFepxnDRFgDJs8ehmC7RwWs

deployed so that `dev` is shown:
https://beoble-toolkit-bdsi47wnx-kryptonium.vercel.app/
https://vercel.com/kryptonium/beoble-toolkit/8HrA2qmQpGRJea27eYJpcoyRgZiW

we should check ASAP after merged to dev, and if dev shows the dev API key related data, we can merge to main and also check ASAP that main show prod API key related data

File: packages/example/pages/_document.tsx [hunk 1]
```diff
@@ -25,15 +25,15 @@ export default class CustomDocument extends Document<{
       <Html>
         <Head>
           {this.props.styleTags}
-          {!process.env.NEXT_PUBLIC_VERCEL_ENV ||
-          process.env.NEXT_PUBLIC_VERCEL_ENV === 'local' ? (
+          {!process.env.VERCEL_ENV ||
+          process.env.VERCEL_ENV === 'development' ? (
```
Comment (bamnenim-kryptonium) [inline packages/example/pages/_document.tsx:29]: this should be seperated bc the src in the script under here is for local testing only.
`development` will affect to our app under dev. domain

**Review by bamnenim-kryptonium — CHANGES_REQUESTED** [review]

File: packages/example/pages/_document.tsx (hunk 1 above)
Comment (vanilla-james) [reply packages/example/pages/_document.tsx:29]: yup as I pointed out in the comment
VERCEL_ENV should be checked against `production`, `preview`, or `development`:

> VERCEL_ENV | The Environment that the app is deployed and running on. The value can be either production, preview, or development.

image

so `development` is local, `preview` is development, and `production` is production.
There is no`local` value, so I replaced with `development`, which is Vercel's local enviroment.

ref:
https://vercel.com/docs/concepts/projects/environment-variables/system-environment-variables#system-environment-variables

so the code as changed above will:
* use 
```
            <script
              id="beoble_test_local"
              data-plugin-key={DEV_PLUGIN_KEY}
              src="http://localhost:1226/plugin/beoble-loader.js"
              async
            />
```
for local (which is `development` in Vercel)

* use 
```
           <script
              id="beoble_test"
              data-plugin-key={PROD_PLUGIN_KEY}
              src="https://cdn.beoble.app/plugin/beoble-loader.js"
              async
            />
```
for production

* use
```
            <script
              id="beoble_test"
              data-plugin-key={DEV_PLUGIN_KEY}
              src="https://dev.cdn.beoble.app/plugin/beoble-loader.js"
              async
            />
```
for non-production and non-local (what we refer to as dev/development stage)

TLDR;
```
VERCEL `development` => our "local"
VERCEL `preview` => our "development" 
VERCEL `production` => our "production" 
```

Comment (bamnenim-kryptonium) [reply packages/example/pages/_document.tsx:29]: Got it thanks for detailed explanation!

**Review by bamnenim-kryptonium — APPROVED** [review]

File: packages/example/pages/_document.tsx (hunk 1 above)
Comment (vanilla-james) [reply packages/example/pages/_document.tsx:29]: > @bamnenim-kryptonium I tested the deployment as much as I could without actually deploying to dev (as deploying to dev is blocked)
> 
> ### How issue was resolved
> * swtiched `process.env.NEXT_PUBLIC_VERCEL_ENV` to `process.env.VERCEL_ENV` as _document.tsx apparently is run in the server and `process.env.NEXT_PUBLIC_VERCEL_ENV` might not behave as we expect / intended
> * `process.env.VERCEL_ENV` should be checked against `production`, `preview`, or `development` - fixed
> 
> ### Test deployments
> deployed so that `prod` is shown: https://beoble-toolkit-9l29p020y-kryptonium.vercel.app/ https://vercel.com/kryptonium/beoble-toolkit/Jq9ydAFepxnDRFgDJs8ehmC7RwWs
> 
> deployed so that `dev` is shown: https://beoble-toolkit-bdsi47wnx-kryptonium.vercel.app/ https://vercel.com/kryptonium/beoble-toolkit/8HrA2qmQpGRJea27eYJpcoyRgZiW
> 
> we should check ASAP after merged to dev, and if dev shows the dev API key related data, we can merge to main and also check ASAP that main show prod API key related data

once merged and deployed we should check if dev is showing dev db data properly, and also the same after merging to main to check if main/production is showing main/production db data properly

Comment (vanilla-james) [reply packages/example/pages/_document.tsx:29]: @bamnenim-kryptonium 
should I merge or do you want to go ahead with it?
